src/components/portfolio/ImagePortfolio.jsx

```diff
@@ -0,0 +1,96 @@
+import DeleteIcon from '../../assets/icons/delete.png';
+import FormUploadPortfolio from './FormUploadPortfolio';
+import Loader from '../common/Loader';
+import useMediaUploader from '../../util/PortfolioUploadMediaService';
+import { useState } from 'react';
+
+const ImagePortfolio = ({ profilePortfolioImages }) => {
+  const {
+    showFileInput,
+    handleButtonClick,
+    handleFileChange,
+    handleMediaSubmit,
+    handleDeleteMedia,
+    error,
+    isLoading,
+  } = useMediaUploader('Images');
+
+  const [selectedImage, setSelectedImage] = useState(null);
+
+  const handleImageClick = (imageUrl) => {
+    setSelectedImage(imageUrl);
+  };
+
+  const handleCloseModal = () => {
+    setSelectedImage(null);
+  };
+  
+  if (isLoading) return <Loader />;
+
+  return (
+    <div className="h-3/5 flex flex-col items-center bg-pureWhite mb-4 p-4">
+      <h2 className="font-spartan font-semibold text-2xl py-2">Portfolio</h2>
+      {profilePortfolioImages.length > 0 ? (
+        <div className="flex flex-wrap gap-6 w-full">
+          {profilePortfolioImages.map((image) => (
+            <div key={image.publicId} className="relative w-28 h-28">
+              <picture>
+                <source srcSet={image.url} type="image/avif" />
+                <source srcSet={image.url} type="image/webp" />
+                <img
+                src={image.url}
+                alt="portfolio image"
+                className="w-full h-full object-cover transition-transform duration-300 ease-in-out hover:scale-110 cursor-pointer"
+                onClick={() => handleImageClick(image.url)}
+              />
+              </picture>
+              <button
+                aria-label="delete image"
+                onClick={() => handleDeleteMedia(image.publicId)}
+                className="absolute top-[-0.5rem] right-[-0.5rem] p-1 hover:border-2 hover:border-red hover:rounded-full"
+              >
+                <img src={DeleteIcon} alt="delete icon" />
```

Comment:
Stretch goal: I noticed when you click on the delete there is no indication that a request has been made. This is fine for now but it would be nice to show the user that the image is being deleted or to at least disable the delete button.

Reply:
Thanks for the comment, added the loading state for delete functions.

Reply:
Are you able to instead disable the delete icon while the delete request is being made. This will indicate that the request has been made and will prevent spamming of the button. This way the functionality of deleting multiple images at once isn't obscured by the loading circle. Ideally you want each image rendering the loading circle within its container when the delete request is being made. **Do not do this now.** This might be a little involved, just a design tip to keep in mind for the future.